miniapps/materials/synthetic_materials.cpp

```diff
@@ -0,0 +1,281 @@
+  double e3 = 0;
+  double pl1 = 1.0;
+  double pl2 = 1.0;
+  double pl3 = 1.0;
+  bool paraview_export = true;
+  bool glvis_export = true;
+
+  OptionsParser args(argc, argv);
+  args.AddOption(&order, "-o", "--order",
+                 "Finite element order (polynomial degree) or -1 for"
+                 " isoparametric space.");
+  args.AddOption(&num_refs, "-r", "--refs", "Number of uniform refinements");
+  args.AddOption(&num_parallel_refs, "-rp", "--refs-parallel",
+                 "Number of uniform refinements");
+  args.AddOption(&topological_support, "-top", "--topology",
+                 "Topological support. 0 particles, 1 octet-truss");
+  args.AddOption(&nu, "-nu", "--nu", "Fractional exponent nu (smoothness)");
+  args.AddOption(&tau, "-t", "--tau", "Parameter for topology generation");
+  args.AddOption(&zeta, "-z", "--zeta",
+                 "Parameter to scale the mixing of topology and randomness");
+  args.AddOption(&l1, "-l1", "--l1",
+                 "First component of diagonal core of theta");
+  args.AddOption(&l2, "-l2", "--l2",
+                 "Second component of diagonal core of theta");
+  args.AddOption(&l3, "-l3", "--l3",
+                 "Third component of diagonal core of theta");
+  args.AddOption(&e1, "-e1", "--e1", "First euler angle for rotation of theta");
+  args.AddOption(&e2, "-e2", "--e2",
+                 "Second euler angle for rotation of theta");
+  args.AddOption(&e3, "-e3", "--e3", "Third euler angle for rotation of theta");
+  args.AddOption(&pl1, "-pl1", "--pl1", "Length scale 1 of particles");
+  args.AddOption(&pl2, "-pl2", "--pl2", "Length scale 2 of particles");
+  args.AddOption(&pl3, "-pl3", "--pl3", "Length scale 3 of particles");
+  args.AddOption(&number_of_particles, "-n", "--number-of-particles",
+                 "Number of particles");
+  args.AddOption(&paraview_export, "-pvis", "--paraview-visualization",
+                 "-no-pvis", "--no-paraview-visualization",
+                 "Enable or disable ParaView visualization.");
+  args.AddOption(&glvis_export, "-gvis", "--glvis-visualization", "-no-gvis",
+                 "--no-glvis-visualization",
+                 "Enable or disable GLVis visualization.");
+  args.Parse();
+  if (!args.Good()) {
+    args.PrintUsage(cout);
+    return 1;
+  }
+  if (Mpi::Root()) {
+    args.PrintOptions(cout);
+  }
+
+  // 2. Read the mesh from the given mesh file.
+  Mesh mesh(mesh_file, 1, 1);
+  int dim = mesh.Dimension();
+
+  // 4. Refine the mesh to increase the resolution.
+  for (int i = 0; i < num_refs; i++) {
+    mesh.UniformRefinement();
+  }
+  ParMesh pmesh(MPI_COMM_WORLD, mesh);
+  mesh.Clear();
+  for (int i = 0; i < num_parallel_refs; i++) {
+    pmesh.UniformRefinement();
+  }
+
+  // 5. Define a finite element space on the mesh.
+  H1_FECollection fec(order, dim);
+  ParFiniteElementSpace fespace(&pmesh, &fec);
+  if (Mpi::Root()) {
+    cout << "Number of finite element unknowns: " << fespace.GetTrueVSize()
```

Comment:
This should be `fespace.GlobalTrueVSize();`

Reply:
fixed in [f0693bd](https://github.com/mfem/mfem/pull/3065/commits/f0693bd5ee7c12012d4356c4d22af9fea3993b55)